src/WebJobs.Extensions.DurableTask/Listener/TaskActivityShim.cs

```diff
@@ -54,7 +54,28 @@ public override async Task<string> RunAsync(TaskContext context, string rawInput
                 functionType: FunctionType.Activity,
                 isReplay: false);
 
-            FunctionResult result = await this.executor.TryExecuteAsync(triggerInput, CancellationToken.None);
+            FunctionResult result;
+            try
+            {
+                result = await this.executor.TryExecuteAsync(triggerInput, CancellationToken.None);
+            }
+            catch (Exception e)
+            {
+                this.config.TraceHelper.FunctionFailed(
+                    this.config.Options.HubName,
+                    this.activityName,
+                    instanceId,
+                    $"An internal error occurred while attempting to execute this function: " + e,
+                    functionType: FunctionType.Activity,
+                    isReplay: false);
+
+                // TODO: Throw a different exception that can be used to abort the orchestration rather than failing it.
```

Comment:
This new exception has to be defined in DurableTask.Core right?

Reply:
Right. I've drafted a proposal for this here: https://github.com/Azure/azure-functions-durable-extension/issues/1133.